File: 1359. Count All Valid Pickup and Delivery Options/count-all-valid-pickup-and-delivery-options.cs
// 1359. Count All Valid Pickup and Delivery Options (3/6/2022)
// Runtime: 31 ms (81.08%) Memory: 25.25 MB (67.57%) 

public class Solution {
    public int CountOrders(int n) {
        long ans = 1;
        long mod = 1000000007;
        
        for (int i = 2; i <= n; i++) {
            ans *= 2 * i - 1;
            ans %= mod;
        }
        
        for (int i = 2; i <= n; i++) {
            ans *= i;
            ans %= mod;
        }
        
        return (int)ans;
    }
}

File: 1359. Count All Valid Pickup and Delivery Options/count-all-valid-pickup-and-delivery-options.py
# 1359. Count All Valid Pickup and Delivery Options (2/10/54148)
# Runtime: 24 ms (50.00%) Memory: 13.43 MB (55.88%) 

class Solution(object):
    def countOrders(self, n):
        """
        :type n: int
        :rtype: int
        """
        ans = 1
        mod = 1000000007
        
        for i in range(2, n+1):
            ans *= 2 * i - 1
            ans %= mod
        
        for i in range(2, n+1):
            ans *= i
            ans %= mod
        
        return ans
